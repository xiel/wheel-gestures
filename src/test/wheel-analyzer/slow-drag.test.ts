import { recordPhases } from '../helper/recordPhases'

import slowDragRight from '../fixtures/slow-drag-right.json'
import squareMove from '../fixtures/square-move-trackpad.json'

describe('drag', () => {
  it('slow drag', () => {
    expect(recordPhases(slowDragRight.wheelEvents)).toMatchSnapshot()
  })

  it('square move - trackpad (tl -> tr -> br -> bl -> tl)', () => {
    expect(recordPhases(squareMove.wheelEvents)).toMatchSnapshot()
  })
})
