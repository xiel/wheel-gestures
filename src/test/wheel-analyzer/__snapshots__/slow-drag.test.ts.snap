// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`drag slow drag 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -1407,
        11,
        0,
      ],
      "axisVelocity": Array [
        -0.125,
        0.0625,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -1,
        "deltaY": 0,
        "timeStamp": 4367,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": false,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        0,
        193,
      ],
      "wheelType": "user",
    },
  ],
}
`;

exports[`drag square move - trackpad (tl -> tr -> br -> bl -> tl) 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -408,
        77,
        0,
      ],
      "axisVelocity": Array [
        0.022727272727272728,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 0,
        "deltaY": -3,
        "timeStamp": 49225,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": false,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        0,
        206,
      ],
      "wheelType": "user",
    },
  ],
}
`;
